src/components/SharedComponents/UploadStatus/UploadStatus.js

```diff
@@ -1,37 +1,101 @@
 // @flow
 
-import INatIcon from "components/INatIcon";
+import INatIcon from "components/SharedComponents/INatIcon";
 import { View } from "components/styledComponents";
-import * as React from "react";
+import type { Node } from "react";
+import React from "react";
+import { Animated, Easing } from "react-native";
 import CircularProgressBase from "react-native-circular-progress-indicator";
-import colors from "styles/tailwindColors";
+import { useTheme } from "react-native-paper";
 
 type Props = {
-  color: string,
-  completeColor: string,
+  color?: string,
+  completeColor?: string,
   progress: number
 }
 
-const UploadStatus = ( { color, completeColor, progress }: Props ): React.Node => (
-  <View className="relative justify-center items-center">
-    {( progress < 0.05 )
-      ? <INatIcon className="m-1" name="status-saved" color={colors.darkGray} size={50} />
-      : (
-        <>
-          {( progress < 1 )
-            ? <INatIcon className="absolute" name="upload-arrow" color={color} size={25} />
-            : <INatIcon className="absolute" name="checkmark" color={completeColor} size={25} />}
-          <CircularProgressBase
-            value={progress}
-            radius={29}
-            activeStrokeColor={( progress < 1 ) ? color : completeColor}
-            progressValueColor={colors.transparent}
-            maxValue={1}
-            inActiveStrokeOpacity={0}
-            activeStrokeWidth={4}
-          />
-        </>
-      )}
-  </View>
-);
+const UploadStatus = ( { color, completeColor, progress }: Props ): Node => {
+  const theme = useTheme();
+  const defaultColor = theme.colors.primary;
+  const defaultCompleteColor = theme.colors.secondary;
+  const rotateAnimation = new Animated.Value( 0 );
+
+  Animated.loop(
+    Animated.timing( rotateAnimation, {
+      toValue: 1,
+      duration: 10000,
+      easing: Easing.linear,
+      useNativeDriver: true
+    } )
+  ).start( () => {
+    rotateAnimation.setValue( 0 );
+  } );
+
+  const interpolateRotating = rotateAnimation.interpolate( {
+    inputRange: [0, 1],
+    outputRange: ["0deg", "360deg"]
+  } );
+
+  const rotate = {
+    transform: [
+      {
+        rotate: interpolateRotating
+      }
+    ]
+  };
+
+  return (
+    <View className="relative items-center justify-center w-[49px] h-[67px]">
```

Comment:
You would also need to add accessibility props here: `accessible` to make sure this View acts as the accessible element over it's children. Then a good label that also reads out aloud the current progress while uploading and the status in the other case: waiting for upload, finished uploading; maybe some more.